src/Compilers/CSharp/Portable/Declarations/DeclarationTableInput.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed class DeclarationTableInput
+    {
+        public static readonly DeclarationTableInput Empty = new(DeclarationTable.Empty, ImmutableList<(bool isAdd, Lazy<RootSingleNamespaceDeclaration> lazyRoot)>.Empty);
+
+        private readonly DeclarationTable _initialTable;
+        private readonly ImmutableList<(bool isAdd, Lazy<RootSingleNamespaceDeclaration> lazyRoot)> _instructions;
```

Comment:
instead of a 'bool', it might be better to have a kind here.  If we have a bool, we should doc that `!isAdd` is explicitly the same as `is removing`.

Reply:
➡️ This change has been moved to #59392